app/mm-redux/actions/threads.ts

```diff
@@ -0,0 +1,242 @@
+                {
+                    type: ThreadTypes.RECEIVED_THREADS,
+                    data: {
+                        ...userThreadList,
+                        threads: userThreadList?.threads?.map((thread) => ({...thread, is_following: true})) ?? [],
+                        team_id: teamId,
+
+                        // Clear threads only when threads are loaded on launch
+                        removeOldThreads: !before && !after && !unread && since === 0,
+                    },
+                },
+            ];
+
+            if (userThreadList.threads?.length) {
+                const flat = require('array.prototype.flat');
+                await dispatch(
+                    getMissingProfilesByIds([
+                        ...new Set(
+                            flat(userThreadList.threads.map(({participants}) => participants.map(({id}) => id))) as string[],
+                        ),
+                    ]),
+                );
+                getThreadsActions.push(
+                    {
+                        type: UserTypes.RECEIVED_PROFILES_LIST,
+                        data: flat(userThreadList.threads.map(({participants: users}) => users)),
+                    },
+                    {
+                        type: PostTypes.RECEIVED_POSTS,
+                        data: {posts: userThreadList.threads.map(({participants, post}) => ({
+                            ...post,
+                            participants,
+                        }))},
+                    },
+                );
+            }
+
+            dispatch(batchActions(getThreadsActions));
+        }
+
+        return {data: userThreadList};
+    };
+}
+
+export function getThread(userId: string, teamId: string, threadId: string, extended = false) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let thread;
+        try {
+            thread = await Client4.getUserThread(userId, teamId, threadId, extended);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        if (thread) {
+            thread = handleThreadArrived(dispatch, getState, thread, teamId);
+        }
+
+        return {data: thread};
+    };
+}
+
+export function handleThreadArrived(dispatch: DispatchFunc, getState: GetStateFunc, threadData: UserThread, teamId: string) {
+    const thread = {...threadData, is_following: true};
+
+    const state = getState();
+    const currentUserId = getCurrentUserId(state);
+    const currentTeamId = getCurrentTeamId(state);
+
+    dispatch(batchActions([
+        {
+            type: UserTypes.RECEIVED_PROFILES_LIST,
+            data: thread.participants?.filter((user) => user.id !== currentUserId),
+        },
+        {
+            type: PostTypes.RECEIVED_POSTS,
+            data: {posts: [{...thread.post, participants: thread.participants}]},
```

Comment:
not sure why is this needed, can you use `ThreadTypes.RECEIVED_THREAD` to handle the incoming data in the posts reducer?